src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -113,15 +117,21 @@ export class DataSourceSelectable extends React.Component<
       });
   }
 
-  onChange(options) {
+  onChange(options: SelectedDataSourceOption[]) {
     if (!this._isMounted) return;
     const selectedDataSource = options.find(({ checked }) => checked);
 
-    this.setState({
-      selectedOption: [selectedDataSource],
-    });
+    this.setState({ dataSourceOptions: options });
+
+    if (selectedDataSource) {
+      this.setState({
+        selectedOption: [selectedDataSource],
+      });
 
-    this.props.onSelectedDataSources([selectedDataSource]);
+      this.props.onSelectedDataSources([
```

Comment:
nit:
non blocking comment~
```suggestion
      this.props.onSelectDataSources([
```

Reply:
It will be disruptive to change the props, plus I think onSelectedDataSources makes more sense